pallets/pools/src/lib.rs

```diff
@@ -701,12 +707,13 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			pool_id: T::PoolId,
 			tranche_loc: TrancheLoc<T::TrancheId>,
-			amount: T::Balance,
+			new_order: T::Balance,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let tranche_id =
-				Pool::<T>::try_mutate(pool_id, |pool| -> Result<T::TrancheId, DispatchError> {
+			let (tranche_id, old_order) = Pool::<T>::try_mutate(
+				pool_id,
+				|pool| -> Result<(T::TrancheId, T::Balance), DispatchError> {
```

Comment:
Do you think we could move these mutations to a separate function? That would keep the high-level extrinsic more declarative, easier to follow and less noisy.

Reply:
I think in this case, we already have sufficient separation of code. This method now calls into `do_update_invest_order`, which on its own calls into `update_order_amount`. I think adding another level will just make it harder to understand.

Reply:
Fair 💪🏻 I was also reviewing this on the train with a smaller screen so everything looked more packed and harder to follow.